lib/daru/io/exporters/csv.rb

```diff
@@ -37,30 +40,48 @@ class CSV < Base
         #   #  1   2   4
         #
         #   Daru::IO::Exporters::CSV.new(df, "filename.csv", convert_comma: true).call
-        def initialize(dataframe, path, converters: :numeric, headers: nil,
-          convert_comma: nil, **options)
+        def initialize(dataframe, path, converters: :numeric, compression: :infer,
+          headers: nil, convert_comma: nil, **options)
           require 'csv'
 
           super(dataframe)
           @path          = path
           @headers       = headers
+          @compression   = compression
           @convert_comma = convert_comma
           @options       = options.merge converters: converters
         end
 
         def call
-          writer = ::CSV.open(@path, 'w', @options)
-          writer << @dataframe.vectors.to_a unless @headers == false
+          contents = process_dataframe
 
-          @dataframe.each_row do |row|
-            writer << if @convert_comma
-                        row.map { |v| v.to_s.tr('.', ',') }
-                      else
-                        row.to_a
-                      end
+          if compression?(:gzip, '.csv.gz')
+            require 'zlib'
+            ::Zlib::GzipWriter.open(@path) do |gz|
+              contents.each { |content| gz.write(content.to_csv(@options)) }
+              gz.close
+            end
+          else
+            csv = ::CSV.open(@path, 'w', @options)
+            contents.each { |content| csv << content }
+            csv.close
           end
+        end
+
+        private
 
-          writer.close
+        def compression?(algorithm, *formats)
+          @compression == algorithm || formats.any? { |f| @path.end_with?(f) }
+        end
+
+        def process_dataframe
+          [].tap do |result|
+            result << @dataframe.vectors.to_a unless @headers == false
+            @dataframe.map_rows do |row|
+              next result << row.to_a unless @convert_comma
+              result << row.map { |v| v.to_s.tr('.', ',') }
```

Comment:
There was no option in CSV stdlib to change the decimal point to comma. Fixed this with commit 8538770 - `:convert_comma` now affects only float values and not strings that contain dots.